pkg/controller/prpqr_reconciler/prpqr_reconciler.go

```diff
@@ -345,10 +384,56 @@ func jobNameHash(name string) string {
 	return hex.EncodeToString(hasher.Sum(nil))
 }
 
-func generateProwjob(rc injectingResolverClient, defaulter periodicDefaulter, baseCiop *api.Metadata, prpqrName, prpqrNamespace string, spec *v1.ReleaseJobSpec, pr *v1.PullRequestUnderTest) (*prowv1.ProwJob, error) {
+func generateAggregatedProwjobs(rc injectingResolverClient, defaulter periodicDefaulter, baseCiop *api.Metadata, prpqrName, prpqrNamespace string, spec jobSpecWithUID, pr *v1.PullRequestUnderTest, aggregatedCount int, statuses map[string]*v1.PullRequestPayloadJobStatus, statusByJobName map[string]*v1.PullRequestPayloadJobStatus, existingProwjobsByNameHash map[string]*prowv1.ProwJob) []*prowv1.ProwJob {
+	var ret []*prowv1.ProwJob
+
+	for i := 0; i < aggregatedCount; i++ {
+		jobName := fmt.Sprintf("%s-%d", spec.JobName(jobconfig.PeriodicPrefix), i)
+		logger := logrus.WithFields(logrus.Fields{"want-job": jobName})
+
+		if status, exists := statusByJobName[jobName]; exists {
+			logger.WithField("prowjob", status.ProwJob).Debug("Job already present in status")
+			statuses[jobName] = status
+			continue
+		}
+
+		if job, exists := existingProwjobsByNameHash[jobNameHash(jobName)]; exists {
+			logger.WithField("prowjob", job.Name).Debug("Prowjob already exists")
+			statuses[jobName] = &v1.PullRequestPayloadJobStatus{
+				ReleaseJobName: jobName,
+				ProwJob:        job.Name,
+				Status:         job.Status,
+			}
+			continue
+		}
+
+		pj, err := generateProwjob(rc, defaulter, baseCiop, prpqrName, prpqrNamespace, spec, pr, jobName)
```

Comment:
This needs to be optimized: right now `generateProwjob` is an expensive operation: injecting the test involves hitting the configresolver. For aggregated jobs we should only need to do it once, and generate the prowjobs from the structure what we obtain.